lib/srv/desktop/rdp/rdpclient/src/lib.rs

```diff
@@ -1103,24 +1222,47 @@ pub struct SharedDirectoryCreateResponse {
     pub err_code: TdpErrCode,
 }
 
-type CGOSharedDirectoryCreateResponse = SharedDirectoryCreateResponse;
-
+#[allow(dead_code)]
 #[derive(Debug)]
-pub struct SharedDirectoryDeleteRequest {
+pub struct SharedDirectoryListResponse {
     completion_id: u32,
-    directory_id: u32,
-    path: String,
+    err_code: TdpErrCode,
+    fso_list: Vec<FileSystemObject>,
+}
+
+impl From<CGOSharedDirectoryListResponse> for SharedDirectoryListResponse {
+    fn from(cgo: CGOSharedDirectoryListResponse) -> SharedDirectoryListResponse {
+        unsafe {
+            let cgo_fso_list = from_go_array(cgo.fso_list, cgo.fso_list_length);
+            let mut fso_list = vec![];
+            for cgo_fso in cgo_fso_list.into_iter() {
+                fso_list.push(FileSystemObject::from(cgo_fso));
+            }
+
+            SharedDirectoryListResponse {
+                completion_id: cgo.completion_id,
+                err_code: cgo.err_code,
+                fso_list,
+            }
+        }
+    }
 }
 
 #[repr(C)]
-pub struct CGOSharedDirectoryDeleteRequest {
-    pub completion_id: u32,
-    pub directory_id: u32,
-    pub path: *const c_char,
+pub struct CGOSharedDirectoryListResponse {
+    completion_id: u32,
+    err_code: TdpErrCode,
+    fso_list_length: u32,
+    fso_list: *mut CGOFileSystemObject,
 }
 
+pub type CGOSharedDirectoryCreateResponse = SharedDirectoryCreateResponse;
+pub type SharedDirectoryDeleteRequest = SharedDirectoryInfoRequest;
+pub type CGOSharedDirectoryDeleteRequest = CGOSharedDirectoryInfoRequest;
 pub type SharedDirectoryDeleteResponse = SharedDirectoryCreateResponse;
 pub type CGOSharedDirectoryDeleteResponse = SharedDirectoryCreateResponse;
+pub type SharedDirectoryListRequest = SharedDirectoryInfoRequest;
+pub type CGOSharedDirectoryListRequest = CGOSharedDirectoryInfoRequest;
```

Comment:
I noticed that a bunch of these are the same structure, and so could just be defined via type aliases. Possibly something similar is possible on the `Go` side.